### PR TITLE
feat: Add support for creating `efa-only` network interfaces

### PR DESCRIPTION
**Pull request by bryantbiggs**
## Description
- Add support for creating `efa-only` network interfaces
- Update tests to default to AL2023

## Motivation and Context
- Recent feature release by EFA
  - Added in version 5.74 of the provider https://github.com/hashicorp/terraform-provider-aws/pull/39882

## Breaking Changes
- No

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

**Comment by bryantbiggs** [comment]
waiting for 2x p5e.48xlarge instances to come available to do final evaluation

**Comment by bryantbiggs** [comment]
image

File: modules/eks-managed-node-group/main.tf [hunk 1]
```diff
@@ -44,13 +44,14 @@ locals {
...
 
+  # Primary network interface must be EFA, remaining can be EFA or EFA-only
   efa_network_interfaces = [
     for i in range(local.num_network_cards) : {
       associate_public_ip_address = false
       delete_on_termination       = true
       device_index                = i == 0 ? 0 : 1
       network_card_index          = i
-      interface_type              = "efa"
+      interface_type              = var.enable_efa_only ? contains(concat([0], var.efa_indices), i) ? "efa" : "efa-only" : "efa"
```
Comment (antonbabenko) [inline modules/eks-managed-node-group/main.tf:54]: `contains(concat([0], var.efa_indices), i)` looks almost like magic but I guess it works as expected. :)

**Review by antonbabenko — APPROVED** [review]

File: modules/eks-managed-node-group/main.tf (hunk 1 above)
Comment (bryantbiggs) [reply modules/eks-managed-node-group/main.tf:54]: ya, hopefully on the next breaking change some of this can be simplified by changing the defaults 😅

**Comment by antonbabenko** [comment]
This PR is included in version 20.28.0 :tada:

**Pull request merged**